Realm/ObjectServerTests/RLMObjectServerTests.m

```diff
@@ -190,92 +309,104 @@ - (void)testUploadChangesWhenRealmOutOfScope {
         usleep(50000);
         RLMRunChildAndWait();
     } else {
-        RLMRealm *r = [self openRealmForURL:url user:user error:&error];
-        XCTAssertNil(error, @"Error when opening Realm: %@", error);
+        RLMRealm *r = [self openRealmForURL:url user:user];
         // Wait for download to complete.
-        [user waitForDownloadToFinish:url];
+        WAIT_FOR_DOWNLOAD(user, url);
         CHECK_COUNT(OBJECT_COUNT, SyncObject, r);
     }
 }
 
-/// A Realm that was opened before a user logged out should be able to resume uploading if the user logs back in.
-- (void)testLogBackInSameRealmUpload {
+/// If a client logs out, the session should be immediately terminated.
+- (void)testImmediateSessionTerminationWhenLoggingOut {
+    const NSInteger OBJECT_COUNT = 10000;
     NSURL *url = [NSURL URLWithString:@"realm://localhost:9080/~/testBasicSync"];
     // Log in the user.
     RLMSyncUser *user = [self logInUserForCredential:[RLMObjectServerTests basicCredential:self.isParent]
                                               server:[RLMObjectServerTests authServerURL]];
-    NSError *error = nil;
-    RLMRealm *r = [self openRealmForURL:url user:user error:&error];
-    XCTAssertNil(error, @"Error when opening Realm: %@", error);
-
+    // Open the Realm
+    RLMRealm *r = [self openRealmForURL:url user:user];
     if (self.isParent) {
         [r beginWriteTransaction];
-        [r addObject:[[SyncObject alloc] initWithValue:@[@"parent-1"]]];
+        for (NSInteger i=0; i<OBJECT_COUNT; i++) {
+            [r addObject:[[SyncObject alloc] initWithValue:@[[NSString stringWithFormat:@"parent-%@", @(i+1)]]]];
+        }
         [r commitWriteTransaction];
+        [user logOut];
+        CHECK_COUNT(OBJECT_COUNT, SyncObject, r);
+        usleep(50000);
```

Comment:
Why the sleep? If it's actually doing anything, there's a good chance this test will fail spuriously on CI machines under load.